### PR TITLE
Evaluate: SARP - A Static Analysis Tool for Runtime Pallets

### PR DESCRIPTION
**Pull request by niklasp**
[Link to Application](https://github.com/w3f/Grants-Program/blob/master/applications/sarp-basic-functionality.md)
[Link to Application PR](https://github.com/w3f/Grants-Program/pull/1706)
[Link to Delivery of Milestone 1/1](https://github.com/w3f/Grant-Milestone-Delivery/blob/70250f0f8d90c3143294e1ec7a25c0c89133a481/deliveries/sarp-basic-functionality-milestone-1.md)

### CODE REVIEW

File: evaluations/sarp_1_niklasp.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+[`ensure_signed`](https://github.com/scs/MIRAI/blob/db8104d6bba5224ad445502ec46166288349e60c/substrate-examples/pallet_template/src/lib.rs#L135)
+instead of additionally testing
+[`ensure_origin`](https://github.com/bhargavbh/MIRAI/blob/5646e7e2775f1b59bed74285ab1c0a8397218fc5/substrate_examples/incorrect-origin/wrong-origin.rs#L49)
+is called correctly. I understand, that tag analysis is also verified to work in
+the code submitted, but is there a reason why you are testing `ensure_signed` is
+called instead? Does it have to do with your Finding #1 that states: "Certain
+pieces of substrate code lead to crashes of MIRAI."
+
+➡️ The output from running `cargo mirai` is not very descriptive,
+[a descriptive message can be added to the `precondition!` macro](https://github.com/facebookexperimental/MIRAI/blob/957af25ce6cc109093cec369c70fe90a1e2544f9/annotations/src/lib.rs#L625).
```
Comment (masapr) [inline evaluations/sarp_1_niklasp.md:87]: Thanks for the input. We'll keep this in mind, as we move on with the project.

File: evaluations/sarp_1_niklasp.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  milestone 1 fork, your naming convention differs. It would be clearer to stick
+  the the former in my opinion.
+
+## User Documentation Feedback
+
+- The instructions on running the code are missing some parts, see
+  [Running the Prototype Code](#running-the-prototype-code)
+
+- The documentation is repetitve in both files in the Beginning ('##Running')
+  submitted as "User Documentation". An ideal solution would have a
```
Comment (masapr) [inline evaluations/sarp_1_niklasp.md:134]: We'll adjust this in the future

File: evaluations/sarp_1_niklasp.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+- [@bhargavbh mentioned](https://github.com/w3f/Grants-Program/pull/1706#issuecomment-1535936716)
+  that it would be relevant to "document interesting vulnarability classes
+  (which you may find in research phase)". The Documentation Delivery
+  specification is: "b) with (interesting) examples of the vulnerability
+  classes." I cannot find that in the M1 delivery, where the focus is only on
+  the "incorrect origin" and "unsigned transaction" vulnerability classes. The
+  [RFP](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/Static-Analysis-for-Runtime-Pallets.md#project-description-page_facing_up)
+  mentions 3 more classes, the RFP repo has a
+  [description and example for the "arithmetic-overflow" vulnarability class](https://github.com/bhargavbh/MIRAI/tree/main/substrate_examples/arithmetic-overflow)
+  and I assume more could be found in engaging with pallet developers.
```
Comment (masapr) [inline evaluations/sarp_1_niklasp.md:153]: yes, we plan on this at a later stage

**Comment by semuelle** [comment]
Thanks for the evaluation, @niklasp. Very nice work. I'll merge the PR, but feel free to continue the conversation with @masapr  and add to the evaluation as needed. Otherwise, we'll take it from here. 

Someone will be in touch with you, @niklasp, soon.

**Comment by niklasp** [comment]
I would be happy about some comments from the w3f team for feedback and the changes made, before continuing with the evaluation from my side.

**Comment by semuelle** [comment]
I will reach out to you by email.

**Comment by RouvenP** [comment]
hi @niklasp we just transferred the KSM

**Pull request merged**